src/Application/Products/Commands/AddProduct/AddProductCommandHandler.cs

```diff
@@ -10,36 +10,38 @@ namespace CleanCompanyName.DDDMicroservice.Application.Products.Commands.AddProd
 
 internal class AddProductCommandHandler : IRequestHandler<AddProductCommand, ProductDto>
 {
-    private readonly IDateTime _dateService;
     private readonly ILogger _logger;
     private readonly IProductRepository _productRepository;
     private readonly IStockClient _stockClient;
     private readonly IValidator<Product> _validator;
 
     public AddProductCommandHandler(
         IProductRepository productRepository,
-        IDateTime dateService,
         IStockClient stockClient,
         ILogger<AddProductCommandHandler> logger,
         IValidator<Product> validator)
     {
         _productRepository = productRepository;
-        _dateService = dateService;
         _stockClient = stockClient;
         _logger = logger;
         _validator = validator;
     }
 
     public async Task<ProductDto> Handle(AddProductCommand request, CancellationToken cancellationToken)
     {
-        var productToAdd = request.Adapt<Product>();
+        var productToAdd = Product.Create(
+            id: request.Id,
+            sku: request.Sku,
+            title: request.Title,
+            description: request.Description,
+            price: request.Price,
+            createdBy: Guid.NewGuid()); // TODO replace with proper user identity
```

Comment:
This Id should come from the caller, or it should be provided by an injected service with the identity
Leaving it on the command I think is the best option here